File: android-custom-view/CustomViewCourse/app/src/main/java/com/chiclaim/customview/MainActivity.java
package com.chiclaim.customview;

import com.chiclaim.customview.view.ExtendViewActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;


public class MainActivity extends BaseActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void goExtendViews(View view) {
        startActivity(new Intent(this, ExtendViewActivity.class));
    }


    private long currentBackPressedTime;

    @Override
    public void onBackPressed() {
        if (getSupportFragmentManager().getBackStackEntryCount() > 0) {
            getSupportFragmentManager().popBackStack();
            return;
        }
        if (System.currentTimeMillis() - currentBackPressedTime > 2000) {
            currentBackPressedTime = System.currentTimeMillis();
            Toast.makeText(this, R.string.tip_app_exit, Toast.LENGTH_SHORT).show();
        } else {
            super.onBackPressed();
        }
    }

    @Override
    protected boolean showBack() {
        return false;
    }
}

File: android-lib-retrofit/retrofit-sample/app/src/main/java/com/chiclaim/android/retrofit_sample/MainActivity.kt
package com.chiclaim.android.retrofit_sample

import android.content.Intent
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        btn_simple_service.setOnClickListener {
            startActivity(Intent(this, SimpleServiceActivity::class.java))
        }

        btn_dynamic_base_url.setOnClickListener {
            startActivity(Intent(this, DynamicBaseUrlActivity::class.java))
        }

        btn_retrofit_post.setOnClickListener {
            startActivity(Intent(this, RetrofitPostActivity::class.java))
        }

        btn_file_upload.setOnClickListener {
            startActivity(Intent(this, FileUploadActivity::class.java))
        }

        btn_custom_call_adapter.setOnClickListener {
            startActivity(Intent(this, CustomAdapterCallActivity::class.java))
        }

        btn_rxjava_call_adapter.setOnClickListener {
            startActivity(Intent(this, RxJavaRetrofitActivity::class.java))
        }

        btn_coroutine.setOnClickListener {
            startActivity(Intent(this, RetrofitWithCoroutineActivity::class.java))
        }

    }

    override fun showBackMenu() = false
}
